src/main/java/com/ucsb/demonextjsspringtodoapp/controllers/TodoController.java

```diff
@@ -30,13 +30,16 @@
   @Autowired
   private TodoRepository todoRepository;
 
+  @Autowired
+  private AuthControllerAdvice authControllerAdvice;
+
   private ObjectMapper mapper = new ObjectMapper();
 
   @PostMapping(value = "/api/todos", produces = "application/json")
   public ResponseEntity<String> createTodo(@RequestHeader("Authorization") String authorization,
       @RequestBody @Valid Todo todo) throws JsonProcessingException {
-    DecodedJWT jwt = JWT.decode(authorization.substring(7));
-    todo.setUserId(jwt.getSubject());
+    AppUser user = authControllerAdvice.getUser(authorization);
+    todo.setUserId(user.getEmail());
```

Comment:
Talked to Scott separately about this, but we should consider using the `sub` string from a user's Auth0 data instead of their email as an identifier for a todo. While I don't suspect that it will happen in this class, there are certain scenarios that could cause a user's email to change, and we should be prepared for when that happens.

Reply:
Issue #47 created to address this concern